### PR TITLE
APS 1082 - Matcher searches for a space

### PR DESCRIPTION
**Pull request by richpjames**
# Context
[JIRA](https://dsdmoj.atlassian.net/jira/software/c/projects/APS/boards/1328?assignee=62a050ba9f5d480069c97f49&selectedIssue=APS-1082)

# Changes in this PR
This PR makes the move from a matcher searching for a bed to search for a space as detailed [in this epic](https://dsdmoj.atlassian.net/browse/APS-1068).  

Most of the changes are repurposing existing code to be suitable for the new endpoints. 

### Before
image

### After
after

### CODE REVIEW

File: server/@types/ui/index.d.ts
Comment (patrickjfl) [inline server/@types/ui/index.d.ts:404]: Are we confident that this code is not being used at present? Generally I've gone for duplicating code for the v2 functionality, to minimise risk of breaking things. But if we're confident it's not being used then it would be better to replace it like you have.

File: server/utils/matchUtils.test.ts
Comment (patrickjfl) [inline server/utils/matchUtils.test.ts:188]: I've gone for a different approach on my branch, of replacing this function with a call to createQueryString to generate uncoded query params for everything we want to pass to the form page.

File: server/utils/matchUtils.test.ts
Comment (patrickjfl) [inline server/utils/matchUtils.test.ts:252]: again, here I'm not expecting any encoding, I'm hoping to pull startDate, duration, premisesName, premisesId, apType from req.query.

File: server/utils/matchUtils.ts
Comment (patrickjfl) [inline server/utils/matchUtils.ts:207]: nice

File: server/controllers/match/search/spaceSearchController.test.ts
Comment (patrickjfl) [inline server/controllers/match/search/spaceSearchController.test.ts:47]: nice

**Review by patrickjfl — APPROVED** [review]
Great work on complicated ticket. I have some questions about the query params being passed into the link to the booking page but that doesn't need to block merging these changes.

File: server/@types/ui/index.d.ts
Comment (richpjames) [reply server/@types/ui/index.d.ts:404]: The only place it is being used is this search page. This page is quite standalone whereas the booking journey that you've been working on is a bit more intermingled with in-use code so I think we've both chosen the right approach despite them being distinct.

File: server/utils/matchUtils.test.ts
Comment (richpjames) [reply server/utils/matchUtils.test.ts:188]: Yeah I thought you may have done something different here, I had to make some of these changes to please the compiler but thought I might as well change 'bed' to 'space' whilst here. We can mop up and delete this function once your PR is merged.

**Pull request merged**
